File: app/src/main/java/org/lineageos/updater/model/UpdateBaseInfo.java
/*
 * SPDX-FileCopyrightText: 2017 The LineageOS Project
 * SPDX-License-Identifier: Apache-2.0
 */
package org.lineageos.updater.model;

public interface UpdateBaseInfo {
    String getName();

    String getDownloadId();

    long getTimestamp();

    String getType();

    String getVersion();

    String getDownloadUrl();

    long getFileSize();
}

File: app/src/main/java/org/lineageos/updater/model/UpdateBaseInfo.kt
/*
 * SPDX-FileCopyrightText: 2017-2023 The LineageOS Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.lineageos.updater.model

interface UpdateBaseInfo {
    var name: String
    var downloadId: String
    var timestamp: Long
    var type: String
    var version: String
    var downloadUrl: String
    var fileSize: Long
}
